org-roam.el

```diff
@@ -231,15 +252,55 @@ If FILE is not specified, use the current buffer's file-path."
          (f-descendant-of-p (file-truename path)
                             (file-truename org-roam-directory))))))
 
-(defun org-roam--list-files (dir)
-  "Return all Org-roam files located within DIR, at any nesting level.
-Ignores hidden files and directories."
+(defun org-roam--shell-command-files (cmd)
+  "Run CMD in the shell and return a list of files. If no files are found, an empty list is returned."
+  (seq-filter #'s-present? (split-string (shell-command-to-string cmd) "\n")))
+
+(defun org-roam--list-files-search-globs (exts)
+  "Given EXTS, return a list of search globs.
+E.g. (\".org\") => (\"*.org\" \"*.org.gpg\")"
+  (append
+   (mapcar (lambda (ext) (s-wrap (concat "*." ext) "\"")) exts)
+   (mapcar (lambda (ext) (s-wrap (concat "*." ext ".gpg") "\"")) exts)))
+
+(defun org-roam--list-files-rg (dir)
+  "Return all Org-roam files located recursively within DIR, using ripgrep."
+  (let* ((globs (org-roam--list-files-search-globs org-roam-file-extensions))
+         (executable (executable-find "rg"))
+         (command (s-join " " `(,executable ,dir "--files"
+                            ,@(mapcar (lambda (glob) (concat "-g " glob)) globs)))))
+    (if executable
```

Comment:
Will the function call ever reach this branch? It looks like in the main entry point you now already check for the executable using `executable-find`, so perhaps this isn't necessary anymore.